twilight-model/src/gateway/payload/incoming/member_add.rs

```diff
@@ -1,79 +1,92 @@
-use crate::guild::Member;
+use crate::{
+    guild::Member,
+    id::{marker::GuildMarker, Id},
+};
 use serde::{Deserialize, Serialize};
 use std::ops::{Deref, DerefMut};
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct MemberAdd(pub Member);
+pub struct MemberAdd {
+    pub member: Member,
+    pub guild_id: Id<GuildMarker>,
```

Comment:
nit: alphabetical ordering
```suggestion
    pub guild_id: Id<GuildMarker>,
    pub member: Member,
```